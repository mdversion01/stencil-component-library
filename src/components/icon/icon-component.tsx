// src/components/icon/icon-component.tsx

import { Component, Prop, h } from '@stencil/core';

@Component({
  tag: 'icon-component',
  styleUrl: 'icon.scss',
  shadow: false,
})
export class IconComponent {
  @Prop() icon: string = '';
  @Prop() iconMargin: string = '';
  @Prop() size: string = '';
  @Prop() tokenIcon: boolean = false;
  @Prop() iconSize?: number;
  @Prop() svg: boolean = false;
  @Prop() color?: string;

  @Prop() iconAriaLabel?: string;          // ✅ renamed
  @Prop() iconAriaHidden: boolean = true;  // ✅ renamed

  private getDynamicStyle(): { [key: string]: string } | undefined {
    const styleObj: { [key: string]: string } = {};
    if (this.iconSize) {
      styleObj['font-size'] = `${this.iconSize}px`;
    }
    if (this.color) {
      styleObj['color'] = this.color;
    }
    return Object.keys(styleObj).length > 0 ? styleObj : undefined;
  }

  render() {
    const dynamicStyle = this.getDynamicStyle();

    const baseClasses = [
      this.icon,
      this.size,
      this.tokenIcon ? 'token-icon' : '',
      this.iconMargin === 'left' ? 'ms-1' : '',
      this.iconMargin === 'right' ? 'me-1' : '',
    ]
      .filter(Boolean)
      .join(' ');

    return (
      <i
        class={baseClasses}
        aria-hidden={this.iconAriaHidden ? 'true' : 'false'}
        aria-label={this.iconAriaLabel || undefined}
        style={dynamicStyle}
      ></i>
    );
  }
}
