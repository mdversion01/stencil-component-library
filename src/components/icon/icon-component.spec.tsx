// src/components/icon/icon-component.spec.tsx
import { newSpecPage } from '@stencil/core/testing';
import { IconComponent } from './icon-component';

describe('icon-component', () => {
  it('renders with default props', async () => {
    const page = await newSpecPage({
      components: [IconComponent],
      html: `<icon-component icon="fa-check"></icon-component>`,
    });

    const iTag = page.root.querySelector('i');
    expect(iTag).toBeTruthy();
    expect(iTag.getAttribute('class')).toContain('fa-check');
    expect(iTag.getAttribute('aria-hidden')).toBe('true');
    expect(iTag.getAttribute('aria-label')).toBeNull();
    expect(page.root).toMatchSnapshot();
  });

  it('renders with iconMargin left', async () => {
    const page = await newSpecPage({
      components: [IconComponent],
      html: `<icon-component icon="fa-user" icon-margin="left"></icon-component>`,
    });

    const iTag = page.root.querySelector('i');
    expect(iTag.className).toContain('ms-1');
    expect(page.root).toMatchSnapshot();
  });

  it('renders with iconMargin right', async () => {
    const page = await newSpecPage({
      components: [IconComponent],
      html: `<icon-component icon="fa-user" icon-margin="right"></icon-component>`,
    });

    const iTag = page.root.querySelector('i');
    expect(iTag.className).toContain('me-1');
    expect(page.root).toMatchSnapshot();
  });

  it('applies tokenIcon class', async () => {
    const page = await newSpecPage({
      components: [IconComponent],
      html: `<icon-component icon="fa-star" token-icon></icon-component>`,
    });

    const iTag = page.root.querySelector('i');
    expect(iTag.className).toContain('token-icon');
    expect(page.root).toMatchSnapshot();
  });

  it('applies dynamic font-size and color', async () => {
    const page = await newSpecPage({
      components: [IconComponent],
      html: `<icon-component icon="fa-star" icon-size="24" color="red"></icon-component>`,
    });

    const iTag = page.root.querySelector('i');
    const styleAttr = iTag.getAttribute('style') || '';
    expect(styleAttr).toContain('font-size: 24px');
    expect(styleAttr).toContain('color: red');
    expect(page.root).toMatchSnapshot();
  });

  it('sets custom aria-label and aria-hidden=false', async () => {
    const page = await newSpecPage({
      components: [IconComponent],
      html: `<icon-component icon="fa-info" icon-aria-label="Information" icon-aria-hidden="false"></icon-component>`,
    });

    const iTag = page.root.querySelector('i');
    expect(iTag.getAttribute('aria-label')).toBe('Information');
    expect(iTag.getAttribute('aria-hidden')).toBe('false');
    expect(page.root).toMatchSnapshot();
  });
});
