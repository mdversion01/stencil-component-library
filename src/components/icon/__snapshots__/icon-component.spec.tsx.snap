// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon-component applies dynamic font-size and color 1`] = `
<icon-component color="red" icon="fa-star" icon-size="24">
  <i aria-hidden="true" class="fa-star" style="font-size: 24px; color: red;"></i>
</icon-component>
`;

exports[`icon-component applies tokenIcon class 1`] = `
<icon-component icon="fa-star" token-icon="">
  <i aria-hidden="true" class="fa-star token-icon"></i>
</icon-component>
`;

exports[`icon-component renders with default props 1`] = `
<icon-component icon="fa-check">
  <i aria-hidden="true" class="fa-check"></i>
</icon-component>
`;

exports[`icon-component renders with iconMargin left 1`] = `
<icon-component icon="fa-user" icon-margin="left">
  <i aria-hidden="true" class="fa-user ms-1"></i>
</icon-component>
`;

exports[`icon-component renders with iconMargin right 1`] = `
<icon-component icon="fa-user" icon-margin="right">
  <i aria-hidden="true" class="fa-user me-1"></i>
</icon-component>
`;

exports[`icon-component sets custom aria-label and aria-hidden=false 1`] = `
<icon-component icon="fa-info" icon-aria-hidden="false" icon-aria-label="Information">
  <i aria-hidden="false" aria-label="Information" class="fa-info"></i>
</icon-component>
`;
